### ISSUE TITLE
todo: create a miniAOD wf for bParking UL data in 12X and 106X; review use of run2_miniAOD_UL in connection with bParking

### ISSUE THREAD

**Issue by slava77** [cms-sw/cmssw#33665]
There is a plan to produce bParking UL AOD in 10_6_X with updates included in #33589 (and maybe more?). Execution of this miniAOD is expected to be done with `Run2_2018,bParking,run2_miniAOD_UL` modifiers specified/enabled (simultaneously).

Since this data is currently not available, a privately produced file can be used to setup the workflow.

Due to differences in AOD content made in 106X UL production campaign and the upcoming bParking UL AOD, actions for `(run2_miniAOD_UL & bParking)` modifier may be different from `(run2_miniAOD_UL & ~bParking)`.

Depending on the implementation of the workflow in the master/12X branch, some of the updates may be necessary in a follow up to #33589.

@bainbrid

**Comment by slava77** [comment]
assign reconstruction

**Comment by slava77** [comment]
assign pdmv

as a part of this issue, is it possible to get some AOD files in the next few days using CMSSW_10_6_25 for the bParking dataset?
I think that 5-10K events based on wf 136.898 for data (here RECO is made by default, making it AOD would be useful) and 1304.181 for MC.
This is based on workflows with bParking modifier that are available in the matrix in 10_6_X.

**Comment by chayanit** [comment]
Hi @slava77 , sure I'm preparing the requests. Could you remind me where I should change to get AOD for data workflow in cmsDriver command? should I just change RECO to AOD in "datatier", "eventcontent" and "step" attributes?
`cmsDriver.py step3 --conditions auto:run2_data --customise Configuration/DataProcessing/RecoTLR.customisePostEra_Run2_2018 --data --datatier RECO,MINIAOD,DQMIO --era Run2_2018,bParking --eventcontent RECO,MINIAOD,DQM --filein "file:step2.root" --fileout "file:step3.root" --hltProcess reHLT --nThreads 8 --no_exec --number 10 --process reRECO --python_filename step_3_cfg.py --runUnscheduled --scenario pp --step RAW2DIGI,L1Reco,RECO,SKIM:SkimBPark,EI,PAT,ALCA:SiStripCalZeroBias+SiStripCalMinBias+SiStripCalSmallBiasScan+TkAlMinBias+EcalESAlign,DQM:@standardDQM+@ExtraHLT+@miniAODDQM`

**Comment by slava77** [comment]
> Hi @slava77 , sure I'm preparing the requests. Could you remind me where I should change to get AOD for data workflow in cmsDriver command? should I just change RECO to AOD in "datatier", "eventcontent" and "step" attributes?
> `cmsDriver.py step3 --conditions auto:run2_data --customise Configuration/DataProcessing/RecoTLR.customisePostEra_Run2_2018 --data --datatier AOD,MINIAOD,DQMIO --era Run2_2018,bParking --eventcontent AOD,MINIAOD,DQM --filein "file:step2.root" --fileout "file:step3.root" --hltProcess reHLT --nThreads 8 --no_exec --number 10 --process reRECO --python_filename step_3_cfg.py --runUnscheduled --scenario pp --step RAW2DIGI,L1Reco,RECO,SKIM:SkimBPark,EI,PAT,ALCA:SiStripCalZeroBias+SiStripCalMinBias+SiStripCalSmallBiasScan+TkAlMinBias+EcalESAlign,DQM:@standardDQM+@ExtraHLT+@miniAODDQM`

eventcontent (primary) and datatier handle this. I edited in the above `s/RECO/AOD/`

**Comment by chayanit** [comment]
Hello @slava77 , I have submitted both requests through relval mechanism. MC is running https://dmytro.web.cern.ch/dmytro/cmsprodmon/workflows.php?prep_id=CMSSW_10_6_25__BParkingUL_PR33665-ProdZEE_13UP18-00001 
But Data relval has some issue with input RAW. It is now stored on tape. We need to transfer it back to disk and lock at T2 CERN before we can get the production to start.

**Comment by slava77** [comment]
it looks like some data pilots outputs are now available.
So, a pair of workflows similar to 136.88811 (data) and 1325.518 (MC).
I'll give it a try this week.

**Comment by chayanit** [comment]
@slava77 yes, please I'm not successful to run Data workflow through RelVal and I noticed that we have also submitted a pilot so I agree to move on with the existing samples.

**Comment by bainbrid** [comment]
hi @slava77 @chayanit

For my understanding: 

1) [Here](https://github.com/cms-sw/cmssw/issues/33665#issuecomment-852394219) you mention 136.898 (data) and 1304.181 (MC). So you plan to duplicate and modify these workflows to handle UL AOD? They both currently run RECO and PAT, the former in the same step, the latter in separate steps, and both use the `bParking` era. 

2) Then [here](https://github.com/cms-sw/cmssw/issues/33665#issuecomment-883482535) you mention two different wfs, 136.88811 (data) and 1325.518 (MC), both run PAT with the `run2_miniAOD_UL` modifier.

- So you plan to use:
  - (1) to test **UL** wfs with `bParking` and 
  - (2) to run on the AOD outputs of 1) to test (re-mini) wfs using `run2_miniAOD_UL`?
- Please can you update me on the status? Happy to test/check once they are in place.

**Comment by slava77** [comment]
> * So you plan to use:
>   
>   * (1) to test **UL** wfs with `bParking` and
>   * (2) to run on the AOD outputs of 1) to test (re-mini) wfs using `run2_miniAOD_UL`?
> 
> * Please can you update me on the status? Happy to test/check once they are in place.

@bainbrid 
Thank you for checking on the status; this somehow slipped out of my attention in August.

The plan originally was to use 1. in 10_6_X to make AOD files representing UL reco files and then 2. to define PAT-only re-miniAOD workflows. 

By now, I think that we may have centrally produced datasets and 1. becomes just a DAS query. 
Do you have a good reference/pointer to use for data and MC from the proper bParking UL AOD in 10_6_X?

**Comment by bainbrid** [comment]
> Do you have a good reference/pointer to use for data and MC from the proper bParking UL AOD in 10_6_X?

Apologies, I'm not quite sure what you mean here by "good reference/pointer". Do you mean: can I point you to an existing data and MC UL BParking sample processed in 10_6_X?

**Comment by slava77** [comment]
On 9/9/21 5:43 AM, bainbrid wrote:
> Apologies, I'm not quite sure what you mean here by "good
> reference/pointer". Do you mean: can I point you to an existing data and
> MC UL BParking sample processed in 10_6_X?
> 

yes.

**Comment by jpata** [comment]
@bainbrid I lost track here - is this issue still revelant, or should it be closed?
